Rails only logs single lines in production

tahb opened the pull request:
<!-- Do you need to update the changelog? -->

## Changes in this PR

- install lograge to help us overwrite Rails default logs in Production
- add some additional options to log helpful information
- prevent the logging of any request to the `/health_check` endpoint to reduce noise
- change the format from "key value" to JSON for easier parsing of the information locally and within log aggregators
- we keep the conventional Rails logging enabled in the `development` environment so it remains familiar to developers of all experiences. The option to enable it is there if teams choose it
- log in UTC only

This will:

- makes our production logs easier to read and reason about
- rails apps will create less logging volume meaning we will have more rolling history available to us
- ignore health_check requests as they happen very frequently and serve us no value as a team looking to understand the behaviour of the system

## Screenshots of UI changes

### Before
![Screenshot 2021-07-23 at 17 36 31](https://user-images.githubusercontent.com/912473/126813820-c9c339b4-879e-4c81-8a3f-1c2d4a897b4e.png)

### After
![Screenshot 2021-07-23 at 17 36 51](https://user-images.githubusercontent.com/912473/126813823-492dbefc-1317-4c4b-ba48-9a51f677beff.png)
## Next steps

- [ ] Run any necessary [accessibility testing](https://playbook.dxw.com/guides/web-accessibility.html) on this feature.

tahb committed:
Health check requests don't log

These produce a lot of logs that aren't helpful to observer application behaviour for live environments.

tahb committed:
Rails logs include timestamps in UTC

tahb committed:
Log more basic info about the request

`current_user` is a convention that most apps use, allow us to group requests to the same user in a more reliable manner than by IP only.

tahb committed:
Log in JSON format

With a machine readable format we can pipe copy lines and manipulate them for our investigation or use it to query out logs more meaningfully in log aggregators.

tahb committed:
Logging to one line disabled in development

By default Rails should return default logs in development because this is familiar to developers of all experiences.

The main benefits are for having this on in production environments, however it can be turned on in develop if the team choose to.

rjw1 reviewed (approved):
Looks good to me.
We should probably check what papertrail does with JSON logs. (I fear it may not be great but that probably shoudlnt stop us)
Maybe add this config to https://github.com/dxw/dalmatian-test-app test that.

Floppy reviewed (approved):
LGTM

tahb commented:
@rjw1 I've opened a PR into the dalmatian test app to confirm how the apps actually appear in Papertrail https://github.com/dxw/dalmatian-test-app/pull/42

tahb commented:
There was a separate ECS issue when testing this on Dalmatian. I used Heroku to demonstrate that json plays nicely with Papertrail: 
![Screenshot 2021-09-02 at 17 22 30](https://user-images.githubusercontent.com/912473/131881085-8ba5d0f3-7bc6-453b-ab0e-4148118f23f9.png)

tahb committed:
Rails logs only use 1 line per request

Rails logs are arguably noisy, unusuable and hard to read. Lograge replaces Rails default logging.

They also take up quite a lot of space for logging services like Papertrail with no added value.
